src/test/java/teammates/test/cases/webapi/ConfirmFeedbackSessionSubmissionActionTest.java

```diff
@@ -23,13 +50,256 @@ protected String getRequestMethod() {
+                Const.ParamsNames.COURSE_ID, typicalCourse1.getId(),
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, session1InCourse1.getFeedbackSessionName(),
+                Const.ParamsNames.SEND_SUBMISSION_EMAIL, "true",
+        };
+        verifyCanAccess(studentSubmitSessionInCourseParams);
+
+        ______TS("Student intends to submit feedback session only contains instructor questions"
+                + ", should be accessible");
+
+        loginAsStudent(student1InCourse2.googleId);
+
+        String[] studentSubmitInstructorSessionParams = new String[] {
+                Const.ParamsNames.INTENT, Intent.STUDENT_SUBMISSION.toString(),
+                Const.ParamsNames.COURSE_ID, typicalCourse2.getId(),
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, session1InCourse2.getFeedbackSessionName(),
+                Const.ParamsNames.SEND_SUBMISSION_EMAIL, "true",
+        };
+        verifyCanAccess(studentSubmitInstructorSessionParams);
+
+        ______TS("Instructor intends to submit feedback session in other course, should not be accessible");
+
+        loginAsInstructor(instructor1OfCourse1.googleId);
+
+        String[] instructorSubmitSessionInOtherCourseParams = new String[] {
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+                Const.ParamsNames.COURSE_ID, typicalCourse2.getId(),
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, session1InCourse2.getFeedbackSessionName(),
+                Const.ParamsNames.SEND_SUBMISSION_EMAIL, "true",
+        };
+
+        verifyCannotAccess(instructorSubmitSessionInOtherCourseParams);
+
+        ______TS("Instructor intends to submit feedback session in his course, should be accessible");
+
+        loginAsInstructor(instructor1OfCourse2.googleId);
+
+        String[] instructorSubmitSessionInCourseParams = new String[] {
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+                Const.ParamsNames.COURSE_ID, typicalCourse2.getId(),
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, session1InCourse2.getFeedbackSessionName(),
+                Const.ParamsNames.SEND_SUBMISSION_EMAIL, "true",
+        };
+
+        verifyCanAccess(instructorSubmitSessionInCourseParams);
+
+        ______TS("Unknown intent, should be accessible");
+
+        loginAsInstructor(instructor1OfCourse2.googleId);
+
+        String[] unknownIntentParams = new String[] {
+                Const.ParamsNames.INTENT, Intent.FULL_DETAIL.toString(),
+                Const.ParamsNames.COURSE_ID, typicalCourse2.getId(),
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, session1InCourse2.getFeedbackSessionName(),
+                Const.ParamsNames.SEND_SUBMISSION_EMAIL, "true",
+        };
+
+        assertThrows(InvalidHttpParameterException.class, () -> getAction(unknownIntentParams).checkAccessControl());
+    }
+
+    private void useTypicalDataBundle() {
```

Comment:
- Override `prepareTestData()` so that it becomes `@BeforeMethod` for test?
- Put the method before all tests so it is clearer?